scipy/linalg/decomp.py

```diff
@@ -1267,3 +1267,138 @@ def hessenberg(a, calc_q=False, overwrite_a=False, check_finite=True):
     q, info = orghr(a=hq, tau=tau, lo=lo, hi=hi, lwork=lwork, overwrite_a=1)
     _check_info(info, 'orghr (hessenberg)', positive=False)
     return h, q
+
+
+def cdf2rdf(w, v):
+    """
+    Converts complex eigensystems to real eigenvalues in a block diagonal form
+    and the associated real eigenvectors.
+
+    If the eigensystem:
+
+    `w, v = scipy.linalg.eig(X)`
+    or
+    `w, v = numpy.linalg.eig(X)`
+
+    has complex eigenvalues appearing in complex-conjugate pairs, cdf2rdf
+    transforms the system so `w` is in real diagonal form, with 2-by-2 real
+    blocks along the diagonal replacing the complex pairs originally there.
+    The right eigenvectors `v` are transformed so that:
+
+    `v @ w = X @ v`
+
+    continues to hold. The individual columns of `v` are no longer
+    eigenvectors, but each pair of vectors associated with a 2-by-2 block
+    in `w` spans the corresponding invariant vectors.
+
+    Parameters
+    ----------
+    w : (M,) array_like or (..., M) array_like of stacked (M,) arrays
```

Comment:
For the output type you can just mention `array_like` and explain the size options in the description of these for better doc rendering.